modules/l4proxyprotocol/matcher.go

```diff
@@ -46,39 +44,15 @@ func (MatchProxyProtocol) CaddyModule() caddy.ModuleInfo {
 
 // Match returns true if the connection looks like it is using the Proxy Protocol.
 func (m MatchProxyProtocol) Match(cx *layer4.Connection) (bool, error) {
-	p := make([]byte, 4)
-	_, err := io.ReadFull(cx, p)
+	buf := make([]byte, 12) // len(headerV2Prefix)
+	_, err := io.ReadFull(cx, buf)
 	if err != nil {
 		return false, err
 	}
-	if bytes.Equal(p, headerV1Prefix) {
-		return true, nil
-	}
-
-	buf := p[:]
-	bufReader := bufio.NewReader(cx)
-	for i := 1; i <= 8; i++ {
-		// read the next 8 bytes, one byte at a time
-		// since 5th byte being null is valid in v2 header
-		// but is considered EOF by readers
-		b, err := bufReader.ReadByte()
-		if err == nil {
-			buf = append(buf, b)
-			continue
-		}
-
-		if !errors.Is(err, io.ErrUnexpectedEOF) {
-			return false, err
-		}
 
-		// 4 bytes were already read, so i == 1 is the 5th byte
-		if i != 1 {
-			return false, err
-		}
-
-		buf = append(buf, 0x00)
+	if bytes.Equal(buf[:5], headerV1Prefix) {
```

Comment:
To avoid the magic number, I think [bytes.HasPrefix](https://pkg.go.dev/bytes#HasPrefix) might be a better alternative.

Reply:
Good point